web/modules/custom/asset_distribution/asset_distribution.module

```diff
@@ -63,6 +64,20 @@ function asset_distribution_form_rdf_entity_asset_distribution_form_alter(&$form
   if (!$rdf || !$rdf instanceof RdfInterface || !in_array($rdf->bundle(), ['asset_release', 'solution'])) {
     throw new AccessDeniedHttpException('Distributions can only be created in context of a solution or an asset release.');
   }
+  $form_state->set('context_rdf_entity_id', $rdf->id());
+  $form['actions']['submit']['#submit'][] = 'asset_distribution_form_rdf_entity_asset_distribution_form_submit';
+}
+
+/**
+ * Submit handler for rdf_entity_asset_distribution_form.
+ *
+ * Redirects to the overview of the parent entity instead of the entity itself.
+ *
+ * @see asset_distribution_form_rdf_entity_asset_distribution_form_alter()
+ */
+function asset_distribution_form_rdf_entity_asset_distribution_form_submit(&$form, FormStateInterface $form_state) {
+  $parent = Rdf::load($form_state->get('context_rdf_entity_id'));
+  $form_state->setRedirectUrl($parent->toUrl());
```

Comment:
The main reason that the add_asset_distribution tests were failing here is the AddContentBlock.
The construction of the `Add <content/rdf type>` buttons is implemented by passing the route context as a parameter to construct the URLs.
By removing this submission handler, the `\Drupal\rdf_entity\Form\RdfForm::buildForm` was kicking in, redirecting to the distribution overview instead of the solution/release overview. 
On the distribution page, the solution is winning over the release parent because the primary actions concern the solution. This means, that while being on the overview of the distribution, the `Add distribution` menu item, concerns the group, not the parent of the current overview, thus, failing on the title constraint, as there *can* be a distribution with the same name as with one of the distributions of a release.